Docker ci

nmanthey opened the pull request:
*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

nmanthey committed:
tools: add ci scripts

We want to make sure that we do not break the default launch procedure
of KTF. Hence, add it to CI.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
workflow: run ci scripts

Make sure the scripts work as expected.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
docker: not interactive

For CI runs, we do not need interactive docker containers.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

wipawel commented on .github/workflows/docker-ci.yml:
```diff
@@ -0,0 +1,17 @@
+name: Docker Run Test
```
This workflow should replace docker-build.yml.

wipawel commented on tools/ci/launch-test.sh:
```diff
@@ -0,0 +1,37 @@
+cd "$SCRIPTDIR"/../..
+
+# Build project in docker
+make clean docker:boot.iso
```
Might make sense to build with UNITTESTS here

wipawel commented on tools/ci/launch-test.sh:
```diff
@@ -0,0 +1,37 @@
+make clean docker:boot.iso
+
+# Use QEMU to launch the guest
+declare -i STATUS=0
+timeout 10 qemu-system-x86_64 \
+    -cdrom boot.iso \
+    -m 1024 \
+    -no-reboot \
+    -serial stdio \
+    -smp cpus=2 || STATUS=$?
```
Nit: Maybe use make boot ?

wipawel commented on tools/ci/launch-test.sh:
```diff
@@ -0,0 +1,37 @@
+    -smp cpus=2 || STATUS=$?
+
+# Check if the expected exit code happened (124 for timeout)
+if [ "$STATUS" -ne 0 ] && [ "$STATUS" -ne 124 ]
```
KTF may loop idle instead of power off. That would hit the timeout condition. Maybe grep serial output for "All tests done" to detect success condition?

nmanthey committed:
Makefile: use bash as shell

Depending no the system, sh is the default shell. This has disadvantages
when trying to use certain features later. Hence, switch the default to
bash.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
docker: forward user and group as run

To avoid issues with the adduser call in the Dockerfile in Ubuntu, but
still allow sharing image artifacts, we will invoke docker runs with the
correct user ID and group ID. This allows to share the build artifacts
without creating a new user inside the container.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
docker: install kmod

When building artifacts in the docker container, the build system
complained about the missing lsmod command. This change adds the
required package to the docker container, so that the command can be
executed as expected.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
README: add Getting Started

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
cpus: use 2 per default

Currently, there are no use cases for 4 CPUs, hence, use less resources.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
qemu: cpu host only with kvm

In case there is no KVM, do not use the cpu host model for QEMU.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
workflows: drop docker-build

With the added docker-ci.yml workflow, we test whether we can build KTF,
as well as whether we can launch it. Consequently, we do not need to
test building separately anymore.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey commented:
* I use "make boot" by now.
 * Running UNITTESTS failed for my setup, so for now I did not use it.
 * I removed the other build workflow
 * I was not able to forward the output in a file, so I did not implement that yet either. I agree it's not ideal, but it's a start.

wipawel committed:
Merge branch 'mainline' into docker-ci

wipawel reviewed (approved):
Looks and works as a charm! Thanks a lot!
